File: Python/백준/Silver/1406. 에디터/에디터.py
import sys
front, back = list(sys.stdin.readline().rstrip()[:]), []
N = int(sys.stdin.readline())
for _ in range(N):
	inp = sys.stdin.readline().rstrip()
	if inp == 'L':
		if front:
			back.append(front.pop())
	elif inp == 'D':
		if back:
			front.append(back.pop())
	elif inp == 'B':
		if front:
			front.pop()
	else:
		front.append(inp[-1])
while back:
	front.append(back.pop())
print("".join(front))

File: Java/백준/Silver/1406. 에디터/에디터.java
import java.io.*;
import java.util.*;
import java.lang.Math;

public class Main {
    static InputStreamReader reader = new InputStreamReader(System.in);
    static BufferedReader br = new BufferedReader(reader);

    static OutputStreamWriter writer = new OutputStreamWriter(System.out);
    static BufferedWriter bw = new BufferedWriter(writer);
    
    public static void main(String[] args) throws IOException {
        String inp = br.readLine();
        int opCnt = Integer.parseInt(br.readLine());
        Deque<Character> dq = new LinkedList<>();
        Stack<Character> stk = new Stack<>();
        for (int i = 0; i < inp.length(); i++) dq.addLast(inp.charAt(i));

        for (int i = 0; i < opCnt; i++) {
            String ops[] = br.readLine().split(" ");
            if (ops[0].equals("L")) {
                if (!dq.isEmpty()) stk.add(dq.pollLast());
            }
            else if (ops[0].equals("D")) {
                if (!stk.isEmpty()) dq.addLast(stk.pop());
            }
            else if (ops[0].equals("B")) {
                if (!dq.isEmpty()) dq.pollLast();
            }
            else if (ops[0].equals("P")) {
                dq.addLast(ops[1].charAt(0));
            }
        }
        while (!dq.isEmpty()) bw.write(dq.pollFirst());
        while (!stk.isEmpty()) bw.write(stk.pop());
        bw.write("\n");
        bw.flush();
        bw.close();
    }
}